Cmake config is broken.

Hi, I'm one of the maintainers of vcpkg, and we recently received a question about libconfg using cmake.
Issue:
There seems to be this problem with the exported cmake usage after we installed libconfig, it fails to link against `libconfig++`.
Run command `./vcpkg install libconfig:x64-windows --head` and output:
```
libconfig provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(libconfig CONFIG REQUIRED)
    target_link_libraries(main PRIVATE libconfig::libconfig libconfig::libconfig++)
```

For this reason, I made a patch to separate `libconfig` and `libconfig++`. 

However, from the code structure, `libconfig` does not seem to be separated, but is linked to `libconfig++`.
BUt, it was broken.
https://github.com/hyperrealm/libconfig/blob/4f13b7f5152427b5e511a09617ca2d9ff63f7800/lib/CMakeLists.txt#L144
https://github.com/hyperrealm/libconfig/blob/4f13b7f5152427b5e511a09617ca2d9ff63f7800/lib/CMakeLists.txt#L152

Reproduce steps:
`CMakeLists.txt`
```
cmake_minimum_required(VERSION 3.0)
project(test)
add_executable(main main.cpp)
find_package(libconfig CONFIG REQUIRED)
target_link_libraries(main PRIVATE libconfig::libconfig libconfig::libconfig++)
```
Error:
```
CMake Error at CMakeLists.txt:5 (target_link_libraries):
  Target "main" links to:

    libconfig::libconfig++

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```
Related vcpkg issue and PR:
https://github.com/microsoft/vcpkg/issues/30906
https://github.com/microsoft/vcpkg/pull/30980

@hyperrealm

--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -123,16 +123,8 @@ target_include_directories(${libname}++
   PUBLIC "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
   )
 
-install(TARGETS ${libname}
+install(TARGETS ${libname} ${libname}++
     EXPORT libconfigTargets
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
-)
-
-install(TARGETS ${libname}++
-    EXPORT libconfig++Targets
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
@@ -141,20 +133,18 @@ install(TARGETS ${libname}++
 
 
 include(CMakePackageConfigHelpers)
-foreach(target_name libconfig libconfig++)
-  write_basic_package_version_file("${target_name}ConfigVersion.cmake"
+write_basic_package_version_file("libconfigConfigVersion.cmake"
     VERSION ${PACKAGE_VERSION}
     COMPATIBILITY SameMajorVersion
     )
 
-  install(EXPORT ${target_name}Targets
-    FILE "${target_name}Config.cmake"
+install(EXPORT libconfigTargets
+    FILE "libconfigConfig.cmake"
     NAMESPACE libconfig::
     DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libconfig
     )
 
-  install(FILES
-    "${CMAKE_CURRENT_BINARY_DIR}/${target_name}ConfigVersion.cmake"
+install(FILES
+    "${CMAKE_CURRENT_BINARY_DIR}/libconfigConfigVersion.cmake"
     DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/libconfig"
     )
-endforeach()

